File: 初始算法/字符串/实现strStr.java
class Solution {
    public int strStr(String haystack, String needle) {
        int nLen = needle.length();
        if(nLen == 0) {
            return 0;
        }
        int hLen = haystack.length();
        int result = -1;
        for(int i = 0;i < hLen;i++) {
            if(haystack.charAt(i) == needle.charAt(0)) {
                result = i;
                for(int j = 1;j < nLen;j++) {
                    if((i + j) >= hLen || haystack.charAt(i + j) != needle.charAt(j)) {
                        result = -1;
                        break;
                    }
                }
            }
            if(result != -1) {
                break;
            }
        }
        
        return result;
    }
}

File: 数据结构/数据和字符串/字符串/实现Strstr.cpp
#include <string>
using namespace std;


class Solution {
public:
    int strStr(string haystack, string needle) {
        int lenOfNeedle = needle.length();
        if(lenOfNeedle == 0) {
            return 0;
        }
        
        int result = -1;
        int lenOfHaystack = haystack.length();
        
        for(int i = 0;i < lenOfHaystack;i++) {
            // 加此判断 则不会超时，判断 子串长度此时比 余下串长度长，无需继续检索了
            if(i + lenOfNeedle > lenOfHaystack) {
                return -1;
            }
            
            if(haystack[i] == needle[0]) {
                result = i;
                for(int j = 1;j < lenOfNeedle;j++) {
                    if(haystack[i + j] != needle[j]) {
                        result = -1;
                        break;
                    }
                }
            }
            
            if(result != -1) {
                return result;
            }
        }
        
        return result;
        
    }
};